cpp/environment/BondOrder.cc

```diff
@@ -44,53 +44,43 @@ BondOrder::BondOrder(unsigned int n_bins_theta, unsigned int n_bins_phi)
         throw invalid_argument("PI must be greater than dp");
 
     // precompute the bin center positions for t
-    m_theta_array = std::shared_ptr<float>(new float[m_n_bins_theta], std::default_delete<float[]>());
+    m_theta_array.prepare(m_n_bins_theta);
     for (unsigned int i = 0; i < m_n_bins_theta; i++)
     {
         float t = float(i) * m_dt;
         float nextt = float(i + 1) * m_dt;
-        m_theta_array.get()[i] = ((t + nextt) / 2.0);
+        m_theta_array[i] = ((t + nextt) / 2.0);
     }
 
     // precompute the bin center positions for p
-    m_phi_array = std::shared_ptr<float>(new float[m_n_bins_phi], std::default_delete<float[]>());
+    m_phi_array.prepare(m_n_bins_phi);
     for (unsigned int i = 0; i < m_n_bins_phi; i++)
     {
         float p = float(i) * m_dp;
         float nextp = float(i + 1) * m_dp;
-        m_phi_array.get()[i] = ((p + nextp) / 2.0);
+        m_phi_array[i] = ((p + nextp) / 2.0);
     }
 
     // precompute the surface area array
-    m_sa_array = std::shared_ptr<float>(new float[m_n_bins_theta * m_n_bins_phi], std::default_delete<float[]>());
-    memset((void*) m_sa_array.get(), 0, sizeof(float) * m_n_bins_theta * m_n_bins_phi);
-    Index2D sa_i = Index2D(m_n_bins_theta, m_n_bins_phi);
+    m_sa_array.prepare({m_n_bins_theta, m_n_bins_phi});
     for (unsigned int i = 0; i < m_n_bins_theta; i++)
     {
         for (unsigned int j = 0; j < m_n_bins_phi; j++)
         {
             float phi = (float) j * m_dp;
             float sa = m_dt * (cos(phi) - cos(phi + m_dp));
-            m_sa_array.get()[sa_i((int) i, (int) j)] = sa;
+            m_sa_array((int) i, (int) j) = sa;
```

Comment:
Is it still necessary to cast to `int`?
```suggestion
            m_sa_array(i, j) = sa;
```